File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        for i in range(len(numbers)):
            rem = target - numbers[i]
            l = i + 1
            r = len(numbers) - 1
            while l <= r:
                mid = l + (r-l) // 2
                if numbers[mid] == rem:
                    return [i+1, mid+1]
                elif numbers[mid] > rem:
                    r = mid - 1
                else:
                    l = mid + 1

        return []

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        for(int i=0;i<nums.size();i++){
            int rem = target - nums[i];
            int l = i+1, r = nums.size()-1;
            int mid;
            while(l<=r){
                mid = l + (r-l) /2 ;

                if(nums[mid]==rem){
                    ans.push_back(i+1);
                    ans.push_back(mid+1);
                    return ans;
                } else if (nums[mid] > rem) {
                    r = mid-1;
                }
                else{
                    l = mid+1;
                }
            }
        }
        return ans;
    }
};